craft_parts/main.py

```diff
@@ -80,12 +80,19 @@ def _process_parts(options: argparse.Namespace) -> None:
     if not cache_dir:
         cache_dir = BaseDirectory.save_cache_path("craft-parts")
 
+    if options.overlay_base:
+        base_layer_hash = options.overlay_base.encode()
```

Comment:
The base layer hash algorithm is not specified and could be anything that remains constant for a given base. The CLI tool just uses the path to the base for simplicity, but real applications can (and probably should) use an actual digest.

Reply:
Added this note as a comment in the code.